pkg/aws/eni/node.go

```diff
@@ -494,18 +494,21 @@ func (n *Node) ResyncInterfacesAndIPs(ctx context.Context, scopedLog *logrus.Ent
 // GetMaximumAllocatableIPv4 returns the maximum amount of IPv4 addresses
 // that can be allocated to the instance
 func (n *Node) GetMaximumAllocatableIPv4() int {
+	if n == nil {
+		return 0
```

Comment:
I think we'd still want to have the log msg statement. However, since in this case `n` is nil, we'd need to use `log` (the global instance of the logger for the ENI subsystem).

Reply:
make sense to me, as loggerLocked() function is already performing nil check, I think I can just use the same n.loggerLocked()

```
func (n *Node) loggerLocked() *logrus.Entry {
	if n == nil || n.node == nil {
		return log
	}

	return log.WithField("instanceID", n.node.InstanceID())
}
```

Reply:
Actualy, we can move the entire if block up, as the lock is mainly for _getLimitsLocked()_, so nil check _n.k8sObj.Spec.ENI.FirstInterfaceIndex_ will not require any read/write lock